velox/substrait/VeloxToSubstraitPlan.cpp

```diff
@@ -378,57 +350,87 @@ void VeloxToSubstraitPlanConvertor::toSubstrait(
   aggregateRel->mutable_common()->mutable_direct();
 }
 
-void VeloxToSubstraitPlanConvertor::toSubstrait(
+void VeloxToSubstraitPlanConvertor::toSubstraitJoin(
     google::protobuf::Arena& arena,
-    const std::shared_ptr<const core::AbstractJoinNode>& joinNode,
-    ::substrait::JoinRel* joinRel) {
-  std::vector<core::PlanNodePtr> sources = joinNode->sources();
+    const std::shared_ptr<const core::HashJoinNode> joinNode,
+    ::substrait::Rel* rel) {
+  const auto& joinSources = joinNode->sources();
+  auto joinOutputRowType =
+      joinSources[0]->outputType()->unionWith(joinSources[1]->outputType());
+
+  if (joinNode->joinType() == core::JoinType::kLeftSemi) {
```

Comment:
for left semi join, `joinSources[1]->outputType()` should be empty?

Reply:
yes. the output of right table should not be include in join projections if joinType = leftSemi